modules/hm/firefox/preferences/default.nix

```diff
@@ -1,11 +1,18 @@
-{cfg, ...}: let
+{
+  cfg,
+  lib,
+  ...
+}: let
   inherit (cfg.theme.colors) background-darker background foreground;
 in {
   # Quality of life stuff
   "browser.download.useDownloadDir" = false;
   "browser.aboutConfig.showWarning" = false;
   "browser.tabs.firefox-view" = false;
-  "browser.toolbars.bookmarks.visibility" = "never";
+  "browser.toolbars.bookmarks.visibility" = lib.mkMerge [
+    (lib.mkIf cfg.misc.displayBookmarksInToolbar "always")
```

Comment:
An enum type would be much cleaner heere. `mkMerge` yields ambigious types in the result, which I'm not very fond of.